### PR TITLE
feat: add AutoRotateRKE2Certs setting

### PR DESCRIPTION
**Pull request by FrankYang0529**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Solution:**
Add a setting to enable automatical rotating RKE2 certificates.

**Related Issue:**
https://github.com/harvester/harvester/issues/3863

**Test plan:**

### Webhook validation

* Test webhook rejects the request if `auto-rotate-rke2-certs` setting value doesn't follow format.
  * The `enable` field must be boolean.
  * The `expiringInHours` field must be integer and bigger or equal than 1.

### Controller validation

1. Create a Harvester cluster.
2. Set `auto-rotate-rke2-certs` setting value as `'{"enable":true,"expiringInHours":8759}'`. The `8759` is 364 days * 24 hours/day + 23 hours.
3. Apply a container to run `curl` to check certificates.
```sh
cat << EOF > alpine.yaml
apiVersion: v1
kind: Pod
metadata:
  name: alpine-go
  namespace: default
spec:
  containers:
  - image: golang:1.20.13-alpine3.19
    command:
      - /bin/sh
      - "-c"
      - "sleep infinity"
    imagePullPolicy: IfNotPresent
    name: alpine
  restartPolicy: Always
EOF
kubectl apply -f alpine.yaml
```
4. Install `curl` and check certificates. The example `expire date` is `Jan 15 02:50:22 2025 GMT`.
```sh
kubectl exec -it alpine-go -- /bin/sh
/go # apk add curl
/go # curl https://<default/kubernetes service ClusterIP> -lkv
# ...
* Server certificate:
*  subject: CN=kube-apiserver
*  start date: Jan 15 08:40:03 2024 GMT
*  expire date: Jan 15 02:50:22 2025 GMT
*  issuer: CN=rke2-server-ca@1705308003
```
5. Check `rkeConfig` in `fleet-local/local` cluster. For a fresh cluster, the `rkeConfig` is empty.
```sh
kubectl get clusters.provisioning.cattle.io -n fleet-local local -o yaml | yq .spec.rkeConfig
```
6. Wait for 1 hour.
7. Check `rkeConfig` in `fleet-local/local` cluster. There is `rotateCertificates` with `generation: 1`.
```sh
kubectl get clusters.provisioning.cattle.io -n fleet-local local -o yaml | yq .spec.rkeConfig.rotateCertificates
```
8. Check certificates again. The example `expire date` is renewed to `Jan 15 03:37:47 2025`.
```sh
kubectl exec -it alpine-go -- /bin/sh
/go # curl https://<default/kubernetes service ClusterIP> -lkv
# ...
* Server certificate:
*  subject: CN=kube-apiserver
*  start date: Jan 15 08:40:03 2024 GMT
*  expire date: Jan 15 03:37:47 2025 GMT
*  issuer: CN=rke2-server-ca@1705308003
```

### CODE REVIEW

File: pkg/controller/master/setting/auto_rotate_rke2_certs.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	if kubernetesIP == "" {
+		logrus.WithFields(logrus.Fields{
+			"name":              setting.Name,
+			"service.namespace": metav1.NamespaceDefault,
+			"service.name":      "kubernetes",
+			"reconcileAfter":    defaultReconcilAutoRotateRKE2CertsSettingDuration,
+		}).Warn("cluster ip is empty, reconcile setting again")
+		h.settingController.EnqueueAfter(setting.Name, defaultReconcilAutoRotateRKE2CertsSettingDuration)
+		return nil
+	}
```
Comment (ibrokethecloud) [inline pkg/controller/master/setting/auto_rotate_rke2_certs.go]: this is unlikely to happen, we should may be just let this error and reconcile at normal backoff interval

Comment (FrankYang0529) [reply pkg/controller/master/setting/auto_rotate_rke2_certs.go]: Updated it. Thank you.

**Review by ibrokethecloud — APPROVED** [review]
lgtm. thanks.

**Review by bk201 — APPROVED** [review]
LGTM

**Pull request merged**